server/routes.go

```diff
@@ -1451,6 +1452,38 @@ func (s *Server) PsHandler(c *gin.Context) {
 	c.JSON(http.StatusOK, api.ProcessResponse{Models: models})
 }
 
+func (s *Server) TemplateHandler(w http.ResponseWriter, r *http.Request) {
+	var req api.TemplateRequest
+	if err := json.NewDecoder(r.Body).Decode(&req); err != nil {
+		http.Error(w, err.Error(), http.StatusBadRequest)
+		return
+	}
+
+	model, err := GetModel(req.Model)
+	if err != nil {
+		switch {
+		case os.IsNotExist(err):
+			http.Error(w, fmt.Sprintf("model '%s' not found", req.Model), http.StatusNotFound)
+		case err.Error() == "invalid model name":
```

Comment:
Please use a sentinel error or check the type. This is going to break something. Especially since this PR introduces no tests to lock this in; but even if it did... assert it's a specific error, not a string.